Fixes some restore issues

AlexOn1ine opened the pull request:
Fixes #6623

Pawkkie commented:
This isn't behaviour we can just right a test for right?

It looks totally fine to me I just to double check that's not easily feasible before approving.

AlexOn1ine commented:
> This isn't behaviour we can just right a test for right?
> 
> It looks totally fine to me I just to double check that's not easily feasible before approving.

You can test it manually by setting up the exact scenario and reading the debug output produced by `SaveBattlerTarget` and `SaveBattlerAttacker`.

The `savedMove` stuff I removed already has a test. I just changed the implementation. See #6376

Pawkkie commented:
> The savedMove stuff I removed already has a test. I just changed the implementation. See https://github.com/rh-hideout/pokeemerald-expansion/pull/6376

Ahhh okay perfect, that's what I was looking for! Thank you for linking :)
